### PR TITLE
Polishing 4.4 build for t31

### PR DESCRIPTION
**Pull request by matteius**
* Minimal changes to get 4.4 image working without mods

Depends on:
* https://github.com/gtxaspec/prudynt-t/pull/3
* https://github.com/themactep/ingenic-sdk/pull/7

Open Questions:
I am still building this with my own toolchain build that I compiled against the 4.4 headers.   Noticed that the toolchain fragments reference the 3_10 kernel sources.

For example, this diff captures how I built locally to verify my changes (but I am not sure what the effect of the 3.10 toolchain would be).

```
diff --git a/configs/fragments/templates/toolchain.download.fragment b/configs/fragments/templates/toolchain.download.fragment
index 2a9228f1..f297ead7 100644
--- a/configs/fragments/templates/toolchain.download.fragment
+++ b/configs/fragments/templates/toolchain.download.fragment
@@ -1,7 +1,7 @@
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
 BR2_TOOLCHAIN_EXTERNAL_DOWNLOAD=y
-BR2_TOOLCHAIN_EXTERNAL_URL="https://github.com/themactep/thingino-firmware/releases/download/toolchain/thingino-toolchain_xburst1_musl_gcc13-linux-mipsel.tar.gz"
+BR2_TOOLCHAIN_EXTERNAL_URL="https://github.com/matteius/ingenic-t31-zrt-kernel-4.4.94/releases/download/4.4.latest/mipsel-thingino-linux-musl_sdk-buildroot.tar.gz"
 BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_10=y
 BR2_TOOLCHAIN_EXTERNAL_CUSTOM_MUSL=y
 BR2_TOOLCHAIN_EXTERNAL_CXX=y
diff --git a/configs/fragments/toolchain.fragment b/configs/fragments/toolchain.fragment
index b2ae16a8..df19c7ad 100644
--- a/configs/fragments/toolchain.fragment
+++ b/configs/fragments/toolchain.fragment
@@ -2,8 +2,8 @@
 BR2_TOOLCHAIN_EXTERNAL=y
 BR2_TOOLCHAIN_EXTERNAL_CUSTOM=y
 BR2_TOOLCHAIN_EXTERNAL_DOWNLOAD=y
-BR2_TOOLCHAIN_EXTERNAL_URL="https://github.com/themactep/thingino-firmware/releases/download/toolchain/thingino-toolchain_xburst1_musl_gcc13-linux-mipsel.tar.gz"
-BR2_TOOLCHAIN_EXTERNAL_HEADERS_3_10=y
+BR2_TOOLCHAIN_EXTERNAL_URL="https://github.com/matteius/ingenic-t31-zrt-kernel-4.4.94/releases/download/4.4.latest/mipsel-thingino-linux-musl_sdk-buildroot.tar.gz"
+BR2_TOOLCHAIN_EXTERNAL_HEADERS_4_4=y
 BR2_TOOLCHAIN_EXTERNAL_CUSTOM_MUSL=y
 BR2_TOOLCHAIN_EXTERNAL_CXX=y
 BR2_INSTALL_LIBSTDCPP=y

### CODE REVIEW

File: configs/modules/t31l_sc2336_atbm6031_defconfig [hunk 1]
```diff
@@ -3,3 +3,6 @@ BR2_SOC_INGENIC_T31L=y
 BR2_SENSOR_MODEL="sc2336"
 BR2_PACKAGE_WIFI=y
 BR2_PACKAGE_WIFI_ATBM6031=y
+CONFIG_KERNEL_VERSION_4=y
```
Comment (themactep) [inline configs/modules/t31l_sc2336_atbm6031_defconfig:6]: lets do that it in an experimental config. we do not want to break the working 3.10 for now.

Comment (matteius) [reply configs/modules/t31l_sc2336_atbm6031_defconfig:6]: Opps, good catch

File: package/prudynt-t/prudynt-t.mk [hunk 2]
```diff
@@ -5,6 +5,9 @@ PRUDYNT_T_DEPENDENCIES = libconfig thingino-live555 ingenic-sdk thingino-freetyp
 PRUDYNT_T_GIT_SUBMODULES = YES
 
 PRUDYNT_CFLAGS += -DPLATFORM_$(shell echo $(SOC_FAMILY) | tr a-z A-Z)
+ifeq ($(KERNEL_VERSION),4.4)
```
Comment (themactep) [inline package/prudynt-t/prudynt-t.mk:8]: this better be done with a boolean: `ifeq ($(KERNEL_VERSION_4),y)`

File: configs/modules/t31l_sc2336_atbm6031_defconfig [hunk 3]
```diff
@@ -3,3 +3,6 @@ BR2_SOC_INGENIC_T31L=y
...
+KERNEL_VERSION_4=y
+BR2_KERNEL_VERSION_4=y
```
Comment (themactep) [inline configs/modules/t31l_sc2336_atbm6031_defconfig:8]: Do we need all three? `KERNEL_VERSION_4=y` is enough to define the kernel version for makefile and packages.

Also, you need to create a separate profile for kernel 4 version in `config/testing/`, this one should stay for 3.10 as legacy/stable.

**Review by themactep — CHANGES_REQUESTED** [review]
Matt, please check the comments and amend the patch so I could merge it clear. Thank you. 

PS: Great job!

**Comment by matteius** [comment]
Power is back -- I'll open a new PR.

**Comment by matteius** [comment]
Replaced by:  https://github.com/themactep/thingino-firmware/pull/40/files

**Pull request closed without merging**